### PR TITLE
refactor: Refactoring the DEX backend from JavaScript to TypeScript 

### PR DESCRIPTION
**Pull request by zzkouniya**
Refactoring the DEX backend from JavaScript to TypeScript

### CODE REVIEW

File: src/modules/balance/balance_ckb_model.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { ApiModel, ApiModelProperty } from "swagger-express-ts";
+
+@ApiModel({
+  description: "Ckb balance",
+  name: "BalanceCkbModel",
+})
+export default class BalanceCkbModel {
```
Comment (homu-chain) [inline src/modules/balance/balance_ckb_model.ts:7]: Extracting an interface would be helpful for front-end

```ts
// replace template string when TypeScript 4.1 released
type NumericHex = string;

interface IBalanceCkbModel {
  free: NumericHex ;
  occupied: NumericHex ;
  locked_order: NumericHex;
}



class BalanceCkbModel implements IBalanceCkbModel  {
 ...
}
```

Comment (zzkouniya) [reply src/modules/balance/balance_ckb_model.ts:7]: Okay, I'll change it to implements.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,27 @@
-node_modules
-*indexer_data*
+### macOS ###
+# General
+.DS_Store
```
Comment (coldsnap666) [inline .gitignore:3]: `.DS_Store` is committed into this PR.
Also, `.github/workflows/actions.yml` the ci file should remain.

File: Dockerfile [hunk 3]
```diff
@@ -1,10 +1,21 @@
...
+ENV INDEXER_FOLDER_PATH ./indexer_data
+ENV ORDER_LOCK_CODE_HASH 0x9c833b9ebd4259ca044d2c47c5e51b7fc25380b07291e54b248d3808f08ed7fd
+ENV ORDER_LOCK_HASH_TYPE data
+
+ADD package.json .
+ADD tsconfig.json .
+ADD src ./src
+
+RUN npm install -f
+RUN npm install typescript -g
```
Comment (coldsnap666) [inline Dockerfile:15]: Why not add `typescript` as devDeps?

Comment (zzkouniya) [reply Dockerfile:15]: I need a global tsc command to help me compile.

File: .gitignore (hunk 2 above)
Comment (zzkouniya) [reply .gitignore:3]: Okay, I'll fix it.

**Pull request merged**